gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -1405,14 +1406,20 @@ private static void checkContentType(BlobInfo blobInfo) throws IllegalArgumentEx
   BatchResponse apply(BatchRequest batchRequest);
 
   /**
-   * Return a channel for reading the blob's content.
+   * Return a channel for reading the blob's content. The blob's latest generation is read. If the
```

Comment:
Are you suggesting that we should set the storage object's generation with the value from `BlobSourceOptions.generationMatch(long)` option?